src/components/pools/whitelist/WhiteListModal.tsx

```diff
@@ -93,7 +95,21 @@ const WhiteListRow = ({
         value={address}
       />
       <Textfield
-        onChange={(e) => onChangeRow(Number(e.target.value), 'amount', rowIndex)}
+        onChange={(e) => {
+          const amount = Number(e.target.value)
+
+          if (amount < 0) {
+            return
+          }
+
+          const decimalsCount = amount.toString().split('.')[1]?.length ?? 0
+
+          if (decimalsCount > maxDecimals) {
```

Comment:
if the token has 18 decimals, users should be able to type up to 18 decimal places included, right? Asking because I'm not sure, I haven't tested it yet.